uni/lib/model/entities/course.dart

```diff
@@ -21,46 +26,36 @@ class Course {
     this.currentAverage,
   });
 
-  static Course? fromJson(Map<String, dynamic> data) {
-    if (data['cur_id'] == null || data['fest_id'] == 0) {
-      return null;
-    }
-
+  factory Course.fromJson(Map<String, dynamic> json) {
     return Course(
-      id: data['cur_id'] as int,
-      festId: data['fest_id'] as int,
-      name: data['cur_nome'] as String?,
-      currYear: data['ano_curricular'] as String?,
-      firstEnrollment: data['fest_a_lect_1_insc'] as int?,
-      abbreviation: data['abbreviation'] as String?,
-      faculty: data['inst_sigla']?.toString().toLowerCase(),
+      id: json['cur_id'] as int,
+      festId: json['fest_id'] as int?,
+      name: json['cur_nome'] as String?,
+      abbreviation: json['abbreviation'] as String?,
+      currYear: json['ano_curricular'] as String?,
+      firstEnrollment: json['fest_a_lect_1_insc'] as int?,
+      state: json['state'] as String?,
+      faculty: json['inst_sigla'].toString().toLowerCase(),
+      finishedEcts: json['finishedEcts'] as num?,
+      currentAverage: json['currentAverage'] as num?,
     );
   }
```

Comment:
Why do we maintain the factory in this case?

Reply:
There is no way to manipulate the attributes automatically with json_serializable. Since faculty is stored as a lower case string I thought that would be the best way.